File: digital-twin-wldt/app/src/main/java/org/example/model/BloodOxygen.java
package org.example.model;

public record BloodOxygen(
        Double percentage,
        Long timestamp,
        Integer timezone
) {
    public static BloodOxygen defaultBloodOxygen() {
        return new BloodOxygen(
                0.0,
                0L,
                0
        );
    }
}

File: bp-data-source/app/src/main/kotlin/bp/data/source/model/BloodOxygen.kt
package bp.data.source.model

import bp.data.source.Mapper

data class BloodOxygenMeasurement(
    val id: Int? = null,
    val percentage: Double? = null,
    val time: Long? = null,
    val time_zone: Int? = null,
    val uid: String? = null,
)

data class BloodOxygen(
    val percentage: Double? = null,
    val timestamp: Long? = null,
    val timezone: Int? = null,
)

object BloodOxygenMapper: Mapper<BloodOxygenMeasurement, BloodOxygen> {
    override fun map(input: BloodOxygenMeasurement, outputClass: Class<BloodOxygen>): BloodOxygen {
        return BloodOxygen(
            percentage = input.percentage,
            timestamp = input.time,
            timezone = input.time_zone
        )
    }
}